pages/company/visualizations/company_core_contributors.py

```diff
@@ -0,0 +1,291 @@
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    # get date value for first contribution
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+    df = df.sort_values(by="created", axis=0, ascending=True)
+    base = df.iloc[0]["created"]
+
+    date_picker = (
+        dcc.DatePickerRange(
+            id=f"{PAGE}-date-picker-range-{VIZ_ID}",
+            min_date_allowed=base,
+            max_date_allowed=dt.date.today(),
+            clearable=True,
+        ),
+    )
+    return date_picker
+
+
+# callback for Company Affiliation by Github Account Info graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"{PAGE}-{paramter_2}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-{paramter_1}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-{paramter_3}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "start_date"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "end_date"),
+    ],
+    background=True,
+    prevent_initial_call=True,
+)
+def compay_associated_activity_graph(repolist, checks, contributions, contributors, start_date, end_date):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, checks, contributions, contributors, start_date, end_date)
+
+    fig = create_figure(df)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, checks, contributions, contributors, start_date, end_date):
+    """Implement your custom data-processing logic in this function.
+    The output of this function is the data you intend to create a visualization with,
+    requiring no further processing."""
+
+    # convert to datetime objects rather than strings
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+
+    # order values chronologically by COLUMN_TO_SORT_BY date
+    df = df.sort_values(by="created", axis=0, ascending=True)
+
+    # filter values based on date picker
+    if start_date is not None:
+        df = df[df.created >= start_date]
+    if end_date is not None:
+        df = df[df.created <= end_date]
+
+    # groups contributions by countributor id and counts
+    df = df.groupby(["cntrb_id", "email_list"], as_index=False)[["created"]].count()
+
+    # filters out contributors that dont meet the core contribution threshhold
+    df = df[df.created >= contributions]
```

Comment:
I don't understand how this defines a core contributor. maybe we should change the language from "core contributor" to something else? that term is reserved for project maintainers and high-profile members of a community so we should be careful

Reply:
This defines a core contributor as someone who meets the threshold of contributions in the time interval defined by the user. If there is a better term im not opposed. This view is meant to get a view into the contributor breakdowns with people who have done "significant" (dont like that word but you get what I am saying) work on the project

Reply:
Yeah it's like "Contributors Filtered by Contribution Volume" or something. Because Core contributors would be a more nuanced measurement but it would make sense to be interested in any naive interpretation- "I want to see the contributors who have made a non-trivial contribution to the project only"